rakefile_common.rb

```diff
@@ -156,6 +156,15 @@ def instructions(key)
   @instructions[key]
 end
 
+def sources
+  @sources ||= begin
```

Comment:
The extra level from this `begin` is a little surprising, and seems unnecessary. 

Wouldn't this be simpler as just:
```ruby
def sources
  @sources ||= instructions(:sources).map do |src|
    raise "Missing `type` field in source: #{src}" if src[:type].to_s.empty?
    Source::Base.instantiate(src)
  end
end
```